crates/rpc-types/src/net.rs

```diff
@@ -149,7 +149,42 @@ impl From<u8> for Connectedness {
         }
     }
 }
+/// Direction represents the direction of a connection.
+#[repr(u8)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Default)]
+pub enum Direction {
+    /// DirUnknown is the default direction when the direction is not specified.
+    #[default]
+    DirUnknown = 0,
+    /// DirInbound is for when the remote peer initiated the connection.
+    DirInbound = 1,
+    /// DirOutbound is for when the local peer initiated the connection.
+    DirOutbound = 2,
+}
+impl From<u8> for Direction {
+    fn from(num: u8) -> Self {
+        match num {
+            0 => Direction::DirInbound,
+            1 => Direction::DirOutbound,
+            2 => Direction::DirUnknown,
+            _ => Direction::DirInbound,
+        }
+    }
+}
```

Comment:
don't think we can do this, mapping unsupported numbers to inbound doesn't feel right, please convert to tryfrom